### PR TITLE
Tank mod fixes, adds bandit garage

### PR DESCRIPTION
**Pull request by DangerNoodle**
Closes https://github.com/CleverRaven/Cataclysm-DDA/issues/18223

Additionally closes https://github.com/CleverRaven/Cataclysm-DDA/issues/18308

Removes override for the bandit camp, instead adding a new overmap for bandit vehicles.

Additionally fixes some weapon capacities. Autoloading tank cannons are described as having a 5-round autoloader, so this fixes that. I do not recall how much @chaosvolt set the capacities for the ATGM turret a and autocannon. but I have increased those as well, based on my best guess about what vehicle the IFV was based off of.

One problem remains. All of them require loading one at a time, and this is only really an issue with the autocannon due to high capacity.

### CODE REVIEW

File: data/mods/Tanks/items.json [hunk 1]
```diff
@@ -351,7 +351,7 @@
   "recoil": 15,
   "durability": 8,
   "burst": 5,
-  "clip_size": 5,
+  "clip_size": 20,
```
Comment (Xnfinity) [inline data/mods/Tanks/items.json]: iirc' you could hold around 150 bullets in the cargo size they had

Comment (DangerNoodle) [inline data/mods/Tanks/items.json]: Ah, thank you. That will be even harder for players to tolerate loading one at a time, but oh well.

Comment (Xnfinity) [inline data/mods/Tanks/items.json]: Ideally you should try mimic the ammo belts for the other Turrets for this weapon

File: data/mods/Tanks/items.json [hunk 2]
```diff
@@ -422,7 +422,7 @@
   "sight_dispersion": 60,
   "recoil": -20,
   "durability": 8,
-  "clip_size": 1,
+  "clip_size": 5,
```
Comment (Xnfinity) [inline data/mods/Tanks/items.json]: This one had 10 shots but that would be overkill really;

File: data/mods/Tanks/items.json (hunk 1 above)
Comment (DangerNoodle) [inline data/mods/Tanks/items.json]: Maybe. Not used to working on magazines.

File: data/mods/Tanks/items.json (hunk 2 above)
Comment (DangerNoodle) [inline data/mods/Tanks/items.json]: Leaving it at 5 should hopefully be fine then.

**Comment by chaosvolt** [comment]
Hmm? I'd been planning to work on an idea of some sort as an alternative to bandit camps, to showcase bandit vehicles, but I kinda never got to even thinking up ideas. o3o

Seems fine by me, sorta looks like something in between the bandit cabin and camp. I have at least been working on a basement addition for arcana (currently dropbox'd: https://dl.dropboxusercontent.com/u/89427928/mapgen_basements.json ) in case you or others want to add it. Something I'd wanted to do looong before some basements got JSONized.

File: data/mods/Tanks/items.json [hunk 3]
```diff
@@ -353,7 +382,10 @@
   "burst": 5,
   "clip_size": 150,
   "reload": 100,
-  "loudness" : 175
+  "loudness" : 175,
+  "magazines": [
+      [ "223", [ "belt30mm" ] ]
```
Comment (Xnfinity) [inline data/mods/Tanks/items.json]: "30mm" not "223"

Comment (DangerNoodle) [inline data/mods/Tanks/items.json]: Copy-paste failure, sorry about that.

**Comment by DangerNoodle** [comment]
I might be able to, later on.

File: data/mods/Tanks/items.json [hunk 4]
```diff
@@ -422,7 +422,7 @@
   "sight_dispersion": 60,
   "recoil": -20,
   "durability": 8,
-  "clip_size": 1,
+  "clip_size": 5,
   "reload": 100,
   "loudness" : 300,
   "flags": [ "NO_RELOAD" ]
```
Comment (BorkBorkGoesTheCode) [inline data/mods/Tanks/items.json]: NO_RELOAD will cause problems.

Comment (DangerNoodle) [inline data/mods/Tanks/items.json]: This seems to be the way that "no reloading vehicle-only" weapons was enforced now that magazines have been implemented.

I suspect a development mistake, in that case?

Comment (DangerNoodle) [inline data/mods/Tanks/items.json]: Tested further. That seems to be incorrect. Even with magazines assigned, currently weapons with reloading disabled will ignore the flag if they are installed on a vehicle. This is fortunate, as I can see no other way to preserve intended behavior.

**Comment by Midaychi** [comment]
@Rivet-the-Zombie @mugling The Atomic Mini-Tank gained an extra "u" on the layout somehow during this, and it causes a crash till you remove it. Offending vowel is on line 11 of vehicles.json

**Pull request merged**